token-swap/program/src/instruction.rs

```diff
@@ -305,29 +311,53 @@ mod tests {
     use super::*;
 
     #[test]
-    fn test_instruction_deserialization() {
+    fn test_instruction_packing() {
         let fee_numerator: u64 = 1;
         let fee_denominator: u64 = 4;
         let nonce: u8 = 255;
-        let check = SwapInstruction::Initialize(InitializationData {
+        let check = SwapInstruction::Initialize {
             fee_numerator,
             fee_denominator,
             nonce,
-        });
-        let packed = check.serialize().unwrap();
-        let unpacked = SwapInstruction::deserialize(&packed).unwrap();
+        };
+        let packed = check.pack();
+        println!("{:?}", packed);
+        let unpacked = SwapInstruction::unpack(&packed).unwrap();
```

Comment:
It doesn't look like we compare this `packed` against the expected `data` below. I like the patterns for the other instructions better.
Also axe that `println` 😄

Reply:
That's embarrassing :sweat_smile: and that's right, the check was never done.  I'll make it look like the other tests